### PR TITLE
CNV-18399: Guest agent ping probe

### PR DESCRIPTION
**Pull request by sjhala-ccs**
Version(s): 4.12+

Issue: [CNV-18399](https://issues.redhat.com//browse/CNV-18399)

Link to docs preview: [About readiness and liveness probes](https://53249--docspreview.netlify.app/openshift-enterprise/latest/virt/logging_events_monitoring/virt-monitoring-vm-health.html#virt-about-readiness-liveness-probes_virt-monitoring-vm-health)

[Defining a guest agent ping probe](https://53249--docspreview.netlify.app/openshift-enterprise/latest/virt/logging_events_monitoring/virt-monitoring-vm-health.html#virt-define-guest-agent-ping-probe_virt-monitoring-vm-health)

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Review by ousleyp — COMMENTED** [review]
This looks really good, just a couple of very minor suggestions. :)

File: modules/virt-define-guest-agent-ping-probe.adoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+.Prerequisites
+
+* The QEMU guest agent must be installed and enabled on the virtual machine.
+
+.Procedure
+
+. Include details of the guest agent ping probe in the VMI configuration file.
++
+
+.Sample guest agent ping probe
```
[the remark this answers is not in the document]
Comment (sjhala-ccs) [reply modules/virt-define-guest-agent-ping-probe.adoc:24]: The code snippets in this assembly use "sample", so I think I might keep it for consistency.

File: modules/virt-define-guest-agent-ping-probe.adoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+----
+# ...
+spec:
+  readinessProbe:
+    guestAgentPing: {} <1>
+    initialDelaySeconds: 120 <2>
+    periodSeconds: 20 <3>
+    timeoutSeconds: 10 <4>
+    failureThreshold: 3 <5>
+    successThreshold: 3 <6>
```
Comment (geetikakay) [inline modules/virt-define-guest-agent-ping-probe.adoc:35]: Maybe a complete yaml would be better.

File: modules/virt-define-guest-agent-ping-probe.adoc [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+# ...
+----
+<1> The guest agent ping probe to connect to the VMI.
```
Comment (geetikakay) [inline modules/virt-define-guest-agent-ping-probe.adoc:38]: Maybe include what is optional and what is mandatory

File: modules/virt-define-guest-agent-ping-probe.adoc (hunk 2 above)
Comment (geetikakay) [reply modules/virt-define-guest-agent-ping-probe.adoc:35]: that means $ oc create -f <file_name>.yaml --> mentioned in line no 48 is not valid here. 
Should we explicitly mentioned what is supported and the above ticket is in progress

File: modules/virt-define-guest-agent-ping-probe.adoc (hunk 3 above)
Comment (dominikholler) [reply modules/virt-define-guest-agent-ping-probe.adoc:38]: Good idea, so we just have to gather the information to write it similar to https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes

File: modules/virt-define-guest-agent-ping-probe.adoc (hunk 2 above)
Comment (dominikholler) [reply modules/virt-define-guest-agent-ping-probe.adoc:35]: It is already inconsistent in the docs currently. The correct flow would be to do this in the UI, which is more hard to describe. I would just leave it as it is.

File: modules/virt-define-guest-agent-ping-probe.adoc (hunk 3 above)
Comment (dominikholler) [reply modules/virt-define-guest-agent-ping-probe.adoc:38]: Looks good, just for reference: https://github.com/kubevirt/kubevirt/blob/3db34971b2df184de40f88c56cd392f356621cc9/staging/src/kubevirt.io/api/core/v1/defaults.go#L158

**Review by dominikholler — APPROVED** [review]
/lgmt

**Comment by geetikakay** [comment]
lgtm

**Pull request merged**